datumaro/plugins/yolo_format/extractor.py

```diff
@@ -287,11 +315,309 @@ def get_subset(self, name):
+        return osp.join(self._path, Yolo8Path.LABELS_FOLDER_NAME, relative_labels_path)
 
     @classmethod
-    def find_sources(cls, path):
-        return cls._find_sources_recursive(path, ".data", "yolo")
+    def name_from_path(cls, path: str) -> str:
+        """
+        Obtains <image name> from the path like [data/]images/<subset>/<image_name>.ext
+
+        <image name> can be <a/b/c/filename>, so it is
+        more involved than just calling "basename()".
+        """
+        path = cls.localize_path(path)
+
+        parts = split_path(path)
+        if 2 < len(parts) and not osp.isabs(path):
+            path = osp.join(*parts[2:])  # pylint: disable=no-value-for-parameter
+        return osp.splitext(path)[0]
+
+    def _iterate_over_image_paths(
+        self, subset_name: str, subset_images_source: Union[str, List[str]]
+    ):
+        if isinstance(subset_images_source, str):
+            if subset_images_source.endswith(YoloPath.SUBSET_LIST_EXT):
+                yield from super()._iterate_over_image_paths(subset_name, subset_images_source)
+            else:
+                path = osp.join(self._path, self.localize_path(subset_images_source))
+                if not osp.isdir(path):
+                    raise InvalidAnnotationError(f"Can't find '{subset_name}' subset image folder")
+                yield from (
+                    osp.relpath(osp.join(root, file), self._path)
+                    for root, dirs, files in os.walk(path)
+                    for file in files
+                    if osp.isfile(osp.join(root, file))
+                )
+        else:
+            yield from subset_images_source
+
+
+class Yolo8SegmentationExtractor(Yolo8Extractor):
+    def _load_segmentation_annotation(
+        self, parts: List[str], image_height: int, image_width: int
+    ) -> Polygon:
+        label_id = self._parse_field(parts[0], int, "Polygon label id")
+        if label_id not in self._categories[AnnotationType.label]:
+            raise UndeclaredLabelError(str(label_id))
+
+        points = [
+            self._parse_field(
+                value, float, f"polygon point {idx // 2} {'x' if idx % 2 == 0 else 'y'}"
+            )
+            for idx, value in enumerate(parts[1:])
+        ]
+        scaled_points = [
+            value * size for value, size in zip(points, cycle((image_width, image_height)))
+        ]
+        return Polygon(scaled_points, label=label_id)
+
+    def _load_one_annotation(
+        self, parts: List[str], image_height: int, image_width: int
+    ) -> Annotation:
+        if len(parts) > 5 and len(parts) % 2 == 1:
+            return self._load_segmentation_annotation(parts, image_height, image_width)
+        raise InvalidAnnotationError(
+            f"Unexpected field count {len(parts)} in the polygon description. "
+            "Expected odd number > 5 of fields for segment annotation (label, x1, y1, x2, y2, x3, y3, ...)"
+        )
+
+
+class Yolo8ObbExtractor(Yolo8Extractor):
```

Comment:
Consider adding the `NAME` class member (for `CliPlugin` integration) or rename to `Yolo8OrientedBoxesExtractor`. "Obb" is not an established acronym (like, e.g., HTTP), so it's not a good idea to have such naming in the code.

You can check the result in the formats list in the `datum import --help` output.

Reply:
renamed Obb to OrientedBoxes everywhere